### PR TITLE
Prevent auto hide labels when redirect panels are rendered.

### PR DESCRIPTION
**Pull request by natocTo**

### CODE REVIEW

File: src/Tracy/assets/Bar/bar.js [hunk 1]
```diff
@@ -275,8 +275,9 @@
 	};
 
 	Bar.prototype.autoHideLabels = function() {
-		var labels = this.elem.querySelectorAll('.tracy-label');
-		for (var i = labels.length - 1; i >= 0 && this.elem.clientHeight >= 40; i--) {
+		var labels = this.elem.querySelectorAll('.tracy-label');¨
```
Comment (JanTvrdik) [inline src/Tracy/assets/Bar/bar.js]: typos at the end of line

**Comment by JanTvrdik** [comment]
Unfortunately the problem is a bit more complicated. Each `<ul>` needs to be handled independently.

**Comment by JanTvrdik** [comment]
This may work

~~~css
#tracy-debug-bar ul {
    overflow: hidden;
}
~~~

~~~js
forEach(this.elem.children, function (ul) {
    var labels = ul.querySelectorAll('.tracy-label');
    for (var i = labels.length - 1; i >= 0 && ul.clientHeight >= 40; i--) {
        labels.item(i).hidden = true;
    }
});
~~~

**Comment by natocTo** [comment]
Ok, I am going to change it.

File: src/Tracy/assets/Bar/bar.js [hunk 2]
```diff
@@ -276,9 +276,13 @@
 
 	Bar.prototype.autoHideLabels = function() {
 		var labels = this.elem.querySelectorAll('.tracy-label');
-		for (var i = labels.length - 1; i >= 0 && this.elem.clientHeight >= 40; i--) {
-			labels.item(i).hidden = true;
-		}
+
```
Comment (JanTvrdik) [inline src/Tracy/assets/Bar/bar.js]: this above line is pointless

**Comment by dg** [comment]
Why overflow: hidden?

**Comment by JanTvrdik** [comment]
Because otherwise `ul.clientHeight === 0`. It's based on http://stackoverflow.com/questions/8721001/putting-a-border-around-floating-elements / http://www.quirksmode.org/css/clearing.html

**Comment by JanTvrdik** [comment]
Note that this fixed redirects but not ajax. I have the patch for AJAX I'm just not sure whether we want to include it here.

**Comment by dg** [comment]
Yep, because LI has float:left… It's time to change wordaround to solid solution - flex.

**Comment by dg** [comment]
Now it removes all labels in first row and some labels in second row

**Comment by dg** [comment]
Closed by 46134d3c1af9bf0c57c673c897fb4de418455a3c
